backend/src/test/java/com/redhat/cloud/notifications/db/EmailAggregationResourcesTest.java

```diff
@@ -39,55 +39,48 @@ void testAllMethods() {
-                            .chain(aggregations -> aggregationResources.getApplicationsWithPendingAggregation(start, end)
-                                    .invoke(keys -> assertEquals(3, keys.size()))
-                            );
+                .chain(() -> aggregationResources.getEmailAggregation(key, start, end))
+                .invoke(aggregations -> {
+                    assertEquals(2, aggregations.size());
+                    assertTrue(aggregations.stream().map(EmailAggregation::getAccountId).allMatch(ACCOUNT_ID::equals));
+                    assertTrue(aggregations.stream().map(EmailAggregation::getBundleName).allMatch(BUNDLE_NAME::equals));
+                    assertTrue(aggregations.stream().map(EmailAggregation::getApplicationName).allMatch(APP_NAME::equals));
+                    assertEquals(1, aggregations.stream().map(EmailAggregation::getPayload).filter(PAYLOAD1::equals).count());
+                    assertEquals(1, aggregations.stream().map(EmailAggregation::getPayload).filter(PAYLOAD2::equals).count());
+                })
+                .chain(() -> aggregationResources.getApplicationsWithPendingAggregation(start, end))
+                .invoke(keys -> {
+                    assertEquals(4, keys.size());
+                    assertEquals(ACCOUNT_ID, keys.get(0).getAccountId());
+                    assertEquals(BUNDLE_NAME, keys.get(0).getBundle());
+                    assertEquals(APP_NAME, keys.get(0).getApplication());
                 })
+                .chain(() -> aggregationResources.purgeOldAggregation(key, end))
+                .invoke(purged -> assertEquals(2, purged))
+                .chain(() -> aggregationResources.getEmailAggregation(key, start, end))
+                .invoke(aggregations -> assertEquals(0, aggregations.size()))
+                .chain(aggregations -> aggregationResources.getApplicationsWithPendingAggregation(start, end))
+                .invoke(keys -> assertEquals(3, keys.size()))
                 .await().indefinitely();
     }
 
     @Test
     void addEmailAggregationWithConstraintViolations() {
         resourceHelpers.addEmailAggregation(ACCOUNT_ID, BUNDLE_NAME, APP_NAME, null)
-                .invoke(added -> assertFalse(added))
+                .invoke(Assertions::assertFalse)
                 .chain(() -> resourceHelpers.addEmailAggregation(ACCOUNT_ID, BUNDLE_NAME, null, PAYLOAD1))
-                .invoke(added -> assertFalse(added))
+                .invoke(Assertions::assertFalse)
                 .chain(() -> resourceHelpers.addEmailAggregation(ACCOUNT_ID, null, APP_NAME, PAYLOAD1))
-                .invoke(added -> assertFalse(added))
+                .invoke(Assertions::assertFalse)
                 .chain(() -> resourceHelpers.addEmailAggregation(null, BUNDLE_NAME, APP_NAME, PAYLOAD1))
-                .invoke(added -> assertFalse(added))
+                .invoke(Assertions::assertFalse)
                 .await().indefinitely();
```

Comment:
Is this `await()` expected?

Reply:
Yes, they will be replaced with [UniAsserter](https://github.com/quarkusio/quarkus/blob/main/test-framework/junit5-vertx/src/main/java/io/quarkus/test/junit/vertx/UniAsserter.java) with the Quarkus 2 bump. For now, we still have a few `await()` around.